### PR TITLE
fix(error-boundary): remove ErrorBoundaryGroup's not on purpose reset because of isMounted's side effect

### PR DESCRIPTION
**Pull request by manudeli**
related with https://github.com/suspensive/react/pull/92

## Remove ErrorBoundaryGroup's not on purpose reset because of isMounted's side effect

useIsMounted of @toss/react is implemented by useState, but we don't need to use useIsMounted triggering rerender component self when after mount.

### `useIsMounted` of @toss/react

```tsx
import { useEffect, useState } from 'react';

/** @tossdocs-ignore */
export function useIsMounted() {
  const [mounted, setMounted] = useState(false);
  useEffect(() => {
    setMounted(true);
  }, []);
  return mounted;
}

```

before this Pull Request, ErrorBoundaryGroup used useIsMounted to prevent reseting ErrorBoundaryGroup self on first rendering, and after mounting and checked whether parent ErrorBoundaryGroup.resetKey is changed to reset self.

but after this Pull Request is merging, we don't need to make unnecessary rerendering due to useIsMounted.

### In addition, we can prevent unintended ErrorBoundaryGroup resetting by useIsMounted.

[When first I add this ErrorBoundaryGroup in @toss/error-boundary](https://github.com/toss/slash/pull/157), I couldn't consider useIsMounted's rerendering side effect

#### AS-IS
ErrorBoundaryGroup will trigger reset on render because of useIsMounted

image

#### TO-BE
ErrorBoundaryGroup won't trigger reset on render because of removed useIsMounted

image

Check more detail and procedure to add this Pull Request in https://github.com/suspensive/react/pull/92 please

## PR Checklist

- [x] I read and included theses actions below

1. I have read the [Contributing Guide](https://github.com/toss/slash/blob/main/.github/CONTRIBUTING.md)
2. I have written documents and tests, if needed.

### CODE REVIEW

File: packages/react/error-boundary/src/ErrorBoundary.tsx [hunk 1]
```diff
@@ -51,7 +51,6 @@ const initialState: ErrorBoundaryState = {
 
 class BaseErrorBoundary extends Component<ErrorBoundaryProps, ErrorBoundaryState> {
   state = initialState;
-  updatedWithError = false;
```
Comment (manudeli) [inline packages/react/error-boundary/src/ErrorBoundary.tsx:54]: I thought this updatedWithError is unnecessary field, there is no updatedWithError in https://github.com/bvaughn/react-error-boundary/blob/master/src/ErrorBoundary.ts too.

If this is matter, let me know why meaningful please.

File: packages/react/error-boundary/src/ErrorBoundary.tsx [hunk 2]
```diff
@@ -67,31 +66,17 @@ class BaseErrorBoundary extends Component<ErrorBoundaryProps, ErrorBoundaryState
     onError?.(error, info);
   }
 
-  resetState() {
-    this.updatedWithError = false;
-    this.setState(initialState);
-  }
-
```
Comment (manudeli) [inline packages/react/error-boundary/src/ErrorBoundary.tsx:74]: I remove this method. if no updatedWithError field, resetState is not anymore meaningful too.
I thought `this.setState(initialState)` is better to express what this is.

File: packages/react/error-boundary/src/hooks/useIsChanged.ts [hunk 3]
```diff
@@ -0,0 +1,4 @@
+/** @tossdocs-ignore */
+import { usePrevious } from '@toss/react';
+
+export const useIsChanged = (value: unknown) => usePrevious(value) !== value;
```
Comment (manudeli) [inline packages/react/error-boundary/src/hooks/useIsChanged.ts:4]: I made custom hook, `useIsChanged` to check whether value is changed with previous rendering value.

this hook will check parent ErrorBoundaryGroup.resetKey is changed
If parentErrorBoundaryGroup.resetKey is changed, ErrorBoundaryGroup self will change resetkey to propagate to child ErrorBoundaryGroup recursively.

**Review by okinawaa — APPROVED** [review]
LGTM👍🚀

**Pull request merged**